var/spack/repos/builtin/packages/py-distributed/package.py

```diff
@@ -50,22 +51,29 @@ class PyDistributed(PythonPackage):
     depends_on("python@3.6:", when="@2:", type=("build", "run"))
     depends_on("python@3.7:", when="@2021.4.1:", type=("build", "run"))
     depends_on("python@3.8:", when="@2022.2.1:", type=("build", "run"))
-    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools@62.6:", type="build", when="@2023.4.1:")
+    depends_on("py-versioneer@0.28+toml", type="build", when="@2023.4.1:")
 
     # In Spack py-dask+distributed depends on py-distributed, not the other way around.
     # Hence, no need for depends_on("py-dask", ...)
     depends_on("py-click@6.6:", type=("build", "run"))
+    depends_on("py-click@8.0:", type=("build", "run"), when="@2023.4.1:")
     depends_on("py-cloudpickle@0.2.2:", type=("build", "run"), when="@:2.16.0")
     depends_on("py-cloudpickle@1.3.0:", type=("build", "run"), when="@2.17.0:2.20.0")
     depends_on("py-cloudpickle@1.5.0:", type=("build", "run"), when="@2.21.0:")
     depends_on("py-jinja2", type=("build", "run"), when="@2022.2.1:")
+    depends_on("py-jinja2@2.10.3", type=("build", "run"), when="@2023.4.1:")
     depends_on("py-locket@1:", type=("build", "run"), when="@2022.2.1:")
     depends_on("py-msgpack", type=("build", "run"), when="@:2.10.0")
     depends_on("py-msgpack@0.6.0:", type=("build", "run"), when="@2.11.0:")
+    depends_on("py-msgpack@1.0.0:", type=("build", "run"), when="@2023.4.1:")
     depends_on("py-packaging@20.0:", type=("build", "run"), when="@2022.2.1:")
     depends_on("py-psutil@5.0:", type=("build", "run"))
+    depends_on("py-psutil@5.7.0:", type=("build", "run"), when="@2023.4.1:")
     depends_on("py-six", type=("build", "run"), when="@:1")
     depends_on("py-sortedcontainers@:1,2.0.2:", type=("build", "run"))
+    depends_on("py-sortedcontainers@2.0.5:", type=("build", "run"), when="@2023.4.1:")
     depends_on("py-tblib", type=("build", "run"), when="@:2.10.0")
     depends_on("py-tblib@1.6.0:", type=("build", "run"), when="@2.11.0:")
     depends_on("py-toolz@0.7.4:", type=("build", "run"), when="@:2.12.0")
```

Comment:
toolz dep needs to be updated

Reply:
What I mean is that this package claims to require tools 0.7.4+ but it actually required 0.10+

Reply:
I'm lost... on line [75](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/py-distributed/package.py#L75) `py-distributed` requires 0.10+

Reply:
Ooh, I didn't see that because the diff doesn't display those lines without expanding them. Sorry for the noise!